src/libs/ReportActionsUtils.js

```diff
@@ -161,35 +160,18 @@ function getLastVisibleMessageText(reportID, actionsToMerge = {}) {
 /**
  * @param {String} reportID
  * @param {Object} [actionsToMerge]
- * @param {Number} deletedSequenceNumber
- * @param {Number} lastReadSequenceNumber
- * @return {Number}
+ * @returns {String}
  */
-function getOptimisticLastReadSequenceNumberForDeletedAction(reportID, actionsToMerge = {}, deletedSequenceNumber, lastReadSequenceNumber) {
-    // If the action we are deleting is unread then just return the current last read sequence number
-    if (deletedSequenceNumber > lastReadSequenceNumber) {
-        return lastReadSequenceNumber;
-    }
-
-    // Otherwise, we must find the first previous index of an action that is not deleted and less than the lastReadSequenceNumber
+function getLastVisibleActionCreated(reportID, actionsToMerge = {}) {
```

Comment:
There is a field on every report action called `created` which stores the datetime that it was created at. This will return the `created` field for the last visible action. 

<img width="572" alt="image" src="https://user-images.githubusercontent.com/1371996/210120833-3fb6d95f-df40-4959-9a8f-c1b4f2a368fc.png">

Is there a different name you think would be clearer? Maybe `getCreatedForLastVisibleAction`?

Reply:
idk maybe I am overthinking it - I can't think of any improvements on this other than `getLastVisibleActionCreatedTime()` - which doesn't feel like much of an improvement. Naming things is hard 😅

Reply:
Seems like we could DRY up some logic between this function and `getLastVisibleMessageText` by adding a function called `getLastVisibleAction`:

```js
/**
 * @param {String} reportID
 * @param {Object} [actionsToMerge]
 * @returns {Object}
 */
function getLastVisibleAction(reportID, actionsToMerge = {}) {
    const actions = _.toArray(lodashMerge({}, allReportActions[reportID], actionsToMerge));
    const visibleActions = _.filter(actions, action => (!isDeletedAction(action)));
    return _.max(visibleActions, action => moment.utc(action.created).valueOf());
}

function getLastVisibleMessageText(reportID, actionsToMerge = {}) {
    const lastVisibleAction = getLastVisibleAction(reportID, actionsToMerge);
    const htmlText = lodashGet(lastVisibleAction, 'message[0].html', '');
    if (!htmlText) {
        return '';
    }

    const parser = new ExpensiMark();
    const messageText = parser.htmlToText(htmlText);
    return ReportUtils.formatReportLastMessageText(messageText);
}

function getLastVisibleActionCreated(reportID, actionsToMerge = {}) {
    const lastVisibleAction = getLastVisibleAction(reportID, actionsToMerge);
    return _.has(lastVisibleAction, 'created') ? lastVisibleAction.created : DateUtils.currentDBTime();
}
```

Reply:
That's a good idea. I'll do that.

Reply:
Ok. I decided to do it like this: 

1. Added `getLastVisibleAction`
2. Removed  `getLastVisibleActionCreated`
3. Have `getLastVisibleMessageText` call `getLastVisibleAction` 
4. Have `reportActionsView` call `getLastVisibleAction` and use the `created` field from the returned object. 

That feels the cleanest to me and seems to solve feedback from both of you. Let me know what you think!